saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -457,12 +462,14 @@ private AssertingPartyDetails(String entityId, boolean wantAuthnRequestsSigned,
 			}
 			Assert.notNull(singleSignOnServiceLocation, "singleSignOnServiceLocation cannot be null");
 			Assert.notNull(singleSignOnServiceBinding, "singleSignOnServiceBinding cannot be null");
+			Assert.notNull(signingAlgorithmUris, "signingAlgorithmUris cannot be null");
```

Comment:
This should ensure that the list also isn't empty. You can achieve this by calling `Assert.notEmpty` instead. I think it would make sense to change the error message to align with that as well.